pyugrid/read_netcdf.py

```diff
@@ -150,9 +150,11 @@ def load_grid_from_nc_dataset(nc, grid, mesh_name=None, load_data=True):
                 raise ValueError(msg(defs['role']))
             continue
         except KeyError:
-            msg = ("File must include {} variables for {} "
-                   "named in mesh variable.").format
-            raise ValueError(msg(coord_names, defs['role']))
+            if defs['required']:
+                msg = ("File must include {} variables for {} "
+                       "named in mesh variable.").format
+                raise ValueError(msg(coord_names, defs['role']))
+            continue
```

Comment:
well, it's sort-of compliance checking -- if the file is too non-compliant, then it can't be loaded at all.

But I suppose that factoring out the compliance checking would be good -- as long as it doesn't make for too much code duplication. This al started as a hack to get something to work, then I refactored it to be a bit cleaner and more of a compliance checker, but it's by no means done :-) 

I've been doing a bit with trying to write an importer for a non-compliant file, and it's been tricky -- ideally, the loading code would be designed to be flexible enough to be able to be re-used for non-compliant files -- with the user specifying extra info. So maybe we work on that, and that will guide how this code should be structured.

I'll post an issue about that.